collect_app/src/main/java/org/odk/collect/android/activities/InstanceChooserList.java

```diff
@@ -150,49 +146,29 @@ public void onCreate(Bundle savedInstanceState) {
 
         BulkFinalizationViewModel bulkFinalizationViewModel = new BulkFinalizationViewModel(
                 scheduler,
-                instancesDataService
+                instancesDataService,
+                settingsProvider
         );
 
-        DraftsMenuProvider draftsMenuProvider = new DraftsMenuProvider(bulkFinalizationViewModel);
-        addMenuProvider(draftsMenuProvider);
-
         MaterialProgressDialogFragment.showOn(this, bulkFinalizationViewModel.isFinalizing(), getSupportFragmentManager(), () -> {
             MaterialProgressDialogFragment dialog = new MaterialProgressDialogFragment();
             dialog.setMessage("Finalizing drafts...");
             return dialog;
         });
 
-        bulkFinalizationViewModel.getFinalizedForms().observe(this, finalizedForms -> {
-            if (!finalizedForms.isConsumed()) {
-                Pair<Integer, Integer> pair = finalizedForms.getValue();
-                if (pair.getSecond().equals(0)) {
-                    SnackbarUtils.showLongSnackbar(
-                            this.findViewById(android.R.id.content),
-                            getResources().getQuantityString(
-                                    plurals.bulk_finalize_success,
-                                    pair.getFirst(),
-                                    pair.getFirst()
-                            )
-                    );
-                } else if (pair.getFirst().equals(pair.getSecond())) {
-                    SnackbarUtils.showLongSnackbar(
-                            this.findViewById(android.R.id.content),
-                            getResources().getQuantityString(
-                                    plurals.bulk_finalize_failure,
-                                    pair.getFirst(),
-                                    pair.getFirst()
-                            )
-                    );
-                } else {
-                    SnackbarUtils.showLongSnackbar(
-                            this.findViewById(android.R.id.content),
-                            getString(string.bulk_finalize_partial_success, pair.getFirst() - pair.getSecond(), pair.getSecond())
-                    );
-                }
-
-                finalizedForms.consume();
-            }
-        });
+        if (bulkFinalizationViewModel.isEnabled()) {
+            DraftsMenuProvider draftsMenuProvider = new DraftsMenuProvider(this, bulkFinalizationViewModel::finalizeAllDrafts);
+            addMenuProvider(draftsMenuProvider, this);
+            bulkFinalizationViewModel.getDraftsCount().observe(this, draftsCount -> {
+                draftsMenuProvider.setDraftsCount(draftsCount);
```

Comment:
We've previously played around with handling observations in `MenuProviders` i.e. treating them more like a `Fragment` than a `View`. I was finding that I needed a lot of test infrastructure to get this working however and that the component seems to prefer this more `View` like setup.

There's unfortunately no docs around `MenuProvider` yet, so it's hard to know what the intention is.